### DISCUSSION TITLE
How to use flat config in `.eslintrc`?

### DISCUSSION THREAD

**Discussion by azat-io** [eslint/eslint#17123] [General] [answered]
Hi!

I'm trying to extend my [root config](https://github.com/azat-io/eslint-workspace-config/blob/main/.eslintrc) from [other flat config](https://github.com/azat-io/eslint-workspace-config/blob/main/eslint/index.js), but get an error:

```
Error: ESLint configuration in .eslintrc » @project/eslint-config-test is invalid:
	- Property "" is the wrong type (expected object but got `[{"languageOptions":{"ecmaVersion":2023,"sourceType":"module"},"rules":{"no-var":"error"}}]`).
```

Steps to reproduce:

1. Clone repo: https://github.com/azat-io/eslint-workspace-config
2. `pnpm install`
3. `pnpm test`

**Comment by azat-io** [comment, marked as answer]
Okay. I understood, moved to `eslint.config.js`: https://github.com/azat-io/eslint-workspace-config/commit/6cdc74b51127aa061aa579edf0cec34a8d060b3e

**Reply by Rec0iL99** [reply]
@azat-io When using a shareable config that uses flat config then the consumer of that config also needs to be using flat config. You are on the right path forward.

**Comment by azat-io** [comment]
Do you plan support `eslint.config.ts`?

**Reply by Rec0iL99** [reply]
@azat-io This was originally proposed for `.eslintrc` in https://github.com/eslint/eslint/issues/12078 and https://github.com/eslint/rfcs/pull/50, but couldn't reach the approval of the core team.
